Upgrades, fixes, sorts and more!

ufoptg opened the pull request:
This is just a small PR with some little updates... there is lots more but start with something like this see what yous thinking.
just talk to us if heading the right way got lots more

**Changes include**


_**Upgrades**_
_**Chatbot** Upgrade and Fix and to use [PaLM 2](https://ai.google/discover/palm2/) or **Gemini**
~ tested and working_


**_New_**
~ _Added OpenAI Api key and 🌀ʊʄ⊕ք🌀 API key, Openweather API, and Google API key **Buttons** to settings > API Keys_
~ _Added **UFoP Banner** to Check and ban like **Ultroid Bans if Enabled**_
~ _Added **SoundCloud** support for link downloads via YouTube plugin_
~ _Added **weather** plugin_
~ _Added **Custom MarkDown** to **fns**._
~ _Added **Gemini_Helper** to **fns**_
~ _Added to small funcs in **helper**.py  **check_reply_to** function_ 
~ _Added **DALL-E-3XL** img gen_
~ _Added **ChatGPTs_**


**_Fixes_**
_**Sangmata** command to use **.sg(For Name)**  or **.sgu(For Username)**_
_**Setwarns** function_
**_Translator_**
_**ShortURL** function_
**_Strings_**


**OTHER**
_Applied custom markdown to **gcasts_** and **_afk_**
_**Gbans** are now logged in log_channel_

Introduced the Keys class for better readability
_for example. **Keys.LOG_CHANNEL** rather than **udB.get_key("LOG_CHANNEL")_**


Assistant now sends PM and Message to LOG_CHANNEL, When you turn chatbot on also if you have sudo users and they are in the LOG_CHANNEL they can also reply to users via your assistant.


**Example usage of custom markdown**

<code>from pyUltroid.fns.custom_markdown import CustomMarkdown

ultroid_bot.parse_mode = CustomMarkdown()
await ultroid_bot.send_message(
    "@UltroidSupportChat", "Hello, this is a [spoiler](spoiler) message! and this is an emoji without the () around emoji/5971944878815317190 [💠]emoji/5971944878815317190"
)</code>


there is more too... ^^^


~ @TrueSaiyan

ufoptg commented on pyUltroid/_misc/_supporter.py:
Default API-Key added for chatbot bot usage and ban checking.

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update chatbot.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update tools.py

ufoptg committed:
Update _chatactions.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update en.yml

ufoptg committed:
Update tools.py

ufoptg committed:
Update helper.py

ufoptg committed:
Update _chatactions.py

ufoptg committed:
Update chatbot.py

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update tools.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update _chatactions.py

ufoptg committed:
Update chatbot.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update chatbot.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update pylint.yaml

ufoptg committed:
Update .env.sample

ufoptg committed:
Update version.py

ufoptg committed:
Update pylint.yaml

ufoptg committed:
Update misc.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update chatbot.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Create chatgpt.py

xditya commented on plugins/_chatactions.py:
```diff
@@ -28,6 +35,38 @@
+            raise ValueError("Error: banlist must be True")
+
+
+# ------------------------- Huge Thanks to @xtdevs -------------------------#
```
you do get a commit attibute, why add "thanks to myself" everywhere?

ufoptg replied:
Ahh because slowly over time I used Ultroid as a base to edit it. And if I ever make my repo public then yeh thought it was right thing to do

xditya commented on plugins/_chatactions.py:
```diff
@@ -195,30 +258,112 @@ async def DummyHandler(ult):
+                await e.delete()
+
+
+@ultroid_bot.on(events.NewMessage(incoming=True))
```
what is oracleBot ?

ufoptg replied:
OracleBot is a branch of Googles Gemini pro. Or Googles Genrative text. 

Basically it's the biggest chat bot available themed in a sense. 

If you like we can literally called it Ultroid AI and it will introduce itself as Ultroids AI

You can set chatbot bases.

ufoptg replied:
@xditya If you want me to remove Oracle from the PR i can.. 

then another review would be good, anything you want removed or changed from this PR ill do.

xditya commented on plugins/extra.py:
```diff
@@ -83,3 +86,35 @@ async def _(e):
+    chat_id = event.chat_id
+
+    # Get the messages in the chat
+    async for message in nimbus_bot.iter_messages(chat_id):
```
`nimbus_bot` ?

ufoptg replied:
Ahh right thanks for that this is what I said before as I've tested and it all works.. I'll fix this

xditya commented on pyUltroid/__main__.py:
```diff
@@ -18,17 +18,16 @@ def main():
 
     try:
-        from apscheduler.schedulers.asyncio import AsyncIOScheduler
+        pass
```
why try..except pass ?

ufoptg replied:
I can re enable this, but I turned the schedulers off to reduce latency and when and if Ultroids fetch ann is still down .

ufoptg replied:
and pylint kept doing it disabled now

xditya replied:
uhm, resolve it

ufoptg replied:
wth i did.... :S

xditya commented on pyUltroid/_misc/_supporter.py:
```diff
@@ -118,6 +118,7 @@ class Config((object)):
         PM_DATA = os.environ.get("PM_DATA", "ENABLE")
         DEEP_AI = os.environ.get("DEEP_AI", None)
         TAG_LOG = os.environ.get("TAG_LOG", None)
+        UFOPAPI = os.environ.get("UFOPAPI", "h32dfKsl8pQUxsNftkogIuDF32pYTwKluIY8emI1Hs")
```
default API key would mean UFoP bans would be enabled by default for everyone. That wouldn't be required.

ufoptg replied:
Bans are not enabled unless it's truned on the same way Ultroidbans is turned on. with .setdb UFoP_BANS True,

The default key is so if anyone wants to they don't have to hunt down a key.

This key also works with Dalle-3-XL and OpenDalle I can add

xditya commented on pyUltroid/dB/botchat_db.py:
```diff
@@ -24,18 +24,20 @@ def get_who(msg_id):
     else:
         ok["TAG"].update({msg: [chat, user]})
-    return udB.set_key("BOTCHAT", ok)
+    return udB.set_key("TAG_LOGGER", ok)
```
please do not change DB key names. This can break existing bots.

xditya commented on pyUltroid/fns/misc.py:
```diff
@@ -272,7 +272,7 @@ async def get_synonyms_or_antonyms(word, type_of_words):
 
 class Quotly:
-    _API = "https://bot.lyo.su/quote/generate"
+    _API = "https://quoteampi.onrender.com/generate"
```
thanks, that was very much needed

ufoptg replied:
buddah uploaded that change... i was going to set up the api myself but on Hugging face, as free render accounts wind down when not used and ive noticed it wind down a few times.

xditya commented on pyUltroid/fns/openaihelper.py:
```diff
@@ -0,0 +1,239 @@
+# openai
+# fake_useragent
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#
+# Special Credits: @LucifSD02 <Telegram>
```
again, commit attributes exist, why populate code with credits?
(also, missing copyright header)a

ufoptg replied:
i removed this file as in not needed

xditya commented on pyUltroid/fns/openaihelper.py:
```diff
@@ -0,0 +1,239 @@
+import requests
+from fake_useragent import UserAgent
+from PIL import Image, ImageColor, ImageDraw, ImageFilter, ImageFont, ImageOps
+from sqlalchemy import Column, String, UnicodeText
```
why sqlalchemy? Ultroid gives the user flexibility to choose their own database, and that is the reason why we have `udB`, which chooses a DB as required by the user.
using sqlalchemy directly here just invalidates that.

ufoptg replied:
To be honest I don't think it needs this. I can remove the file I'm pretty sure this was an old  file for something else thank you

xditya commented on pyUltroid/fns/tools.py:
```diff
@@ -432,19 +438,142 @@ async def get_google_images(query):
     random.shuffle(google_images)
     return google_images
 
+###############################################################################
```
again, a whole load of credits to myself. why?

ufoptg replied:
In the event I made my repo public, it's basically your bot.

xditya commented on pyUltroid/fns/tools.py:
```diff
@@ -432,19 +438,142 @@ async def get_google_images(query):
+        is_gemini_oracle: bool = False,
+        gemini_api_key: str = None
+    ):
+        url = f"https://ufoptg-ufop-api.hf.space/UFoP/gemini-the-oracle"
```
why not just use the gemini api directly?

ufoptg replied:
Because using gemini API directly doesn't allow for continued conversation.. so when you talk or use the chat bot it doesn't forget whats been said to eachother unless you ask it to purge the chat history.

well so i thought ill read what you linked me
and because it is forcing it into Oracle mode.

The api it is using allows for real time training of the chatbot too. but gemini's api probably the same

xditya commented on pyUltroid/fns/tools.py:
```diff
@@ -432,19 +438,142 @@ async def get_google_images(query):
 
+UFoP_API = udB.get_key("UFOPAPI") 
+###############################################################################
+# By @TrueSaiyan           Huge Thanks to @xditya!!!!!!                       #
```
again, credits

ufoptg replied:
I'll go through and fix it all

ufoptg replied:
I'll go through and fix it all

xditya commented on pyUltroid/version.py:
```diff
@@ -1,2 +1,2 @@
 __version__ = "2023.02.20"
-ultroid_version = "0.8"
+ultroid_version = "0.8.5"
```
gg

ufoptg replied:
Again this wasn't meant to go over I added .5 to the repo Ive made

ufoptg committed:
Update extra.py

ufoptg committed:
Update __main__.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update __main__.py

ufoptg committed:
pylint: auto fixes

ufoptg commented on plugins/_chatactions.py:
```diff
@@ -97,6 +136,30 @@ async def DummyHandler(ult):
             except BaseException:
                 pass
+
+        if udB.get_key("UFoP_BANS"):
```
Yes so default API key is so that if UFoP_BANS = True

ufoptg committed:
Update botchat_db.py

ufoptg committed:
Update botchat_db.py

ufoptg committed:
Update __main__.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update __main__.py

ufoptg committed:
Update helper.py

ufoptg committed:
Delete pyUltroid/fns/openaihelper.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update version.py

ufoptg committed:
Update en.yml

ufoptg reviewed (commented):
Let me know hope thats a bit better lol i see what you were saying now lol

credit mad

ufoptg committed:
Update tools.py

ufoptg committed:
Update _supporter.py

ufoptg committed:
Create chatgpt.py

ufoptg committed:
Update chatgpt.py

ufoptg commented on plugins/chatgpt.py:
let me know if you want this not on here,

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update tools.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update helper.py

ufoptg committed:
Update _supporter.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Delete plugins/chatgpt.py

ufoptg commented on pyUltroid/fns/tools.py:
```diff
@@ -432,19 +438,145 @@ async def get_google_images(query):
+        self,
+        api_key: str = None,
+        user_id: int = None,
+        mongo_url: str = None,
```
It's probably me being lazy, but you can see this using official API 

People can actually just set a mongo db to store their chats 

Or can do it through offical

ufoptg committed:
Update _supporter.py

ufoptg committed:
Update _supporter.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update callbackstuffs.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update helper.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update _chatactions.py

ufoptg committed:
Update _chatactions.py

ufoptg committed:
Merge branch 'TeamUltroid:main' into patch-2

ufoptg committed:
Update globaltools.py

Log gbans and ungbans in log channel

ufoptg committed:
Merge branch 'TeamUltroid:main' into patch-1

ufoptg committed:
Update botchat_db.py

ufoptg committed:
Update google_image.py

ufoptg committed:
Update custom_markdown.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update _supporter.py

ufoptg committed:
Update callbackstuffs.py

ufoptg committed:
Update .env.sample

ufoptg committed:
Merge pull request #2 from ufoptg/patch-1

Patch 1

ufoptg committed:
Merge branch 'test' into patch-2

ufoptg committed:
Merge pull request #3 from ufoptg/patch-2

Patch 2

ufoptg committed:
Update helper.py

ufoptg committed:
Update callbackstuffs.py

ufoptg committed:
Merge pull request #4 from ufoptg/patch-2

Patch 2

ufoptg committed:
Update callbackstuffs.py

ufoptg committed:
Update helper.py

ufoptg committed:
Update .env.sample

ufoptg committed:
auto fixes

ufoptg committed:
fix

ufoptg committed:
auto fixes

ufoptg committed:
Merge pull request #5 from ufoptg/patch-2

Patch 2

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Merge pull request #6 from ufoptg/patch-2

Update chatgpt.py

ufoptg committed:
Merge branch 'test' into patch-1

xditya commented on resources/session/ssgen.py:
```diff
@@ -55,7 +55,6 @@ def get_api_id_and_hash():
 def telethon_session():
     try:
         spinner("tele")
```
where's the import gone ?

xditya commented on strings/__init__.py:
```diff
@@ -42,7 +42,7 @@ def get_string(key: str, _res: bool = True) -> Any:
         try:
             en_ = languages["en"][key]
-            tr = translate(en_, lang_tgt=lang).replace("\ N", "\n")
+            tr = translate(en_, lang_tgt=lang).replace("\\ N", "\n")
```
extra \

xditya commented on pyUltroid/fns/helper.py:
```diff
@@ -286,7 +288,7 @@ async def bash(cmd, run_code=0):
     out = stdout.decode().strip()
     if not run_code and err:
-        if match := re.match("\/bin\/sh: (.*): ?(\w+): not found", err):
+        if match := re.match("\\/bin\\/sh: (.*): ?(\\w+): not found", err):
```
extra \ here

ufoptg replied:
when the lint auto fixed for some reason it didnt register 
if match := re.match("\/bin\/sh: (.*): ?(\w+): not found", err):
as if it had an r"\/bin\/sh: (.*): ?(\w+): not found"

only thing i can possible conclude

ufoptg committed:
CustomMarkdown support for gcasts

gadmincast, gucast, and gcast now all support premium emojis and spoiler messages

ufoptg committed:
pylint: auto fixes

ufoptg committed:
© date correction

ufoptg committed:
Update __main__.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Update ssgen.py

ufoptg committed:
Update __init__.py

ufoptg committed:
Update helper.py

ufoptg committed:
Update callbackstuffs.py

ufoptg committed:
PmChatBot - Pm/LogChatbot

Works in log and Pm

Also any sudo users in log can also reply via bot and use /who

ufoptg committed:
GeminiHelper

tba

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update tools.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update tools.py

ufoptg committed:
Added Soundcloud link support

ytsc downloads soundcloud links

ufoptg committed:
Fix setwarns

ufoptg committed:
Update __main__.py

ufoptg committed:
Update _database.py

ufoptg committed:
Update funcs.py

ufoptg committed:
Update .env.sample

ufoptg committed:
Added Gemini

#TODO 
 create command for function inside GeminiUltroid class to clear mongo db if need be.

ufoptg committed:
Update en.yml

ufoptg committed:
Update en.yml

ufoptg committed:
Create weather.py

ufoptg committed:
Custom MD semi support

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update tools.py

ufoptg committed:
Translator Fix

ufoptg committed:
Webshot fix

ufoptg committed:
Update requirements.txt

ufoptg committed:
translator fix

ufoptg committed:
Update helper.py

ufoptg committed:
imports

ufoptg committed:
Telegraph error handle

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update pmbot.py

ufoptg committed:
Gemini

ufoptg committed:
fix

ufoptg committed:
Security fix

ufoptg commented on plugins/tools.py:
```diff
@@ -438,24 +474,36 @@
-        f"• **Ultroid Tiny**\n• Given Url : {url}\n• Shorten Url : {data['response']['tinyUrl']}"
-    )
+        s = pyshorteners.Shortener()
+        if "http://tinyurl.com" in input_url.lower():
```
resolved

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
gpt string

ufoptg committed:
strings

ufoptg committed:
Update en.yml

ufoptg committed:
Update youtube.py

ufoptg committed:
Update en.yml

ufoptg committed:
Update writer.py

ufoptg committed:
strings

ufoptg committed:
Update words.py

ufoptg committed:
Update words.py

ufoptg committed:
Update webupload.py

ufoptg committed:
Update writer.py

ufoptg committed:
strings

ufoptg committed:
Update videotools.py

ufoptg committed:
Update vctools.py

ufoptg committed:
Update variables.py

ufoptg committed:
Update en.yml

ufoptg committed:
Update utilities.py

ufoptg committed:
Update usage.py

ufoptg committed:
Update unsplash.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update tag.py

ufoptg committed:
Update sudo.py

ufoptg committed:
Update stickertools.py

ufoptg committed:
Update specialtools.py

ufoptg committed:
Update chatbot.py

ufoptg committed:
Chatbot Finalization

ufoptg committed:
strings

ufoptg committed:
Update search.py

ufoptg committed:
Update schedulemsg.py

ufoptg committed:
Update resize.py

ufoptg committed:
Update qrcode.py

ufoptg committed:
Update profile.py

ufoptg committed:
strings

ufoptg committed:
Update __init__.py

ufoptg committed:
Update __init__.py

ufoptg committed:
Update __init__.py

ufoptg committed:
Update __init__.py

ufoptg committed:
Update profanityfilter.py

ufoptg committed:
Update polls.py

ufoptg committed:
Update pmpermit.py

ufoptg committed:
Update en.yml

ufoptg committed:
Update pdftools.py

ufoptg committed:
Update other.py

ufoptg committed:
Update notes.py

ufoptg committed:
Update en.yml

ufoptg committed:
Update nsfwfilter.py

ufoptg committed:
Update en.yml

ufoptg committed:
Update nightmode.py

ufoptg committed:
Update mute.py

ufoptg committed:
Update misc.py

ufoptg committed:
Update mediatools.py

ufoptg committed:
Update logo.py

ufoptg committed:
Update locks.py

ufoptg committed:
Update imagetools.py

ufoptg committed:
Update greetings.py

ufoptg committed:
Update globaltools.py

ufoptg committed:
Update en.yml

ufoptg committed:
Update glitch.py

ufoptg committed:
Update giftools.py

ufoptg committed:
Update gdrive.py

ufoptg committed:
Update forcesubscribe.py

ufoptg committed:
Update extra.py

ufoptg committed:
help strings

ufoptg committed:
Google Search Fix

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update chatgpt.py

invalid-email-address committed:
fix

ufoptg committed:
Merge branch 'patch-1' of https://github.com/ufoptg/Ultroid into patch-1

ufoptg committed:
Update tools.py

ufoptg committed:
Merge branch 'patch-1' of https://github.com/ufoptg/Ultroid into patch-1

ufoptg committed:
Update tools.py

ufoptg committed:
Update chatbot.py

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update _chatactions.py

ufoptg committed:
Update chatgpt.py

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update chatbot.py

ufoptg committed:
pylint: auto fixes

ufoptg committed:
Delete LOCKDENV

ufoptg committed:
Update _chatactions.py

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update _chatactions.py

ufoptg committed:
Update __main__.py

ufoptg committed:
Update __main__.py

ufoptg committed:
Update misc.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update tools.py

ufoptg committed:
shorten and expand url

fixed shorten and expand url

ufoptg committed:
Update __init__.py

ufoptg committed:
Update __init__.py

ufoptg committed:
Update logo.py

ufoptg committed:
Update BaseClient.py

ufoptg committed:
Update tools.py

ufoptg committed:
Update requirements.txt

ufoptg committed:
Update optional-requirements.txt

ufoptg committed:
Update version.py

ufoptg committed:
Update loader.py

ufoptg committed:
Update codeql-analysis.yml

ufoptg committed:
Update __init__.py

ufoptg committed:
Merge branch 'main' of https://github.com/TeamUltroid/Ultroid into TeamUltroid-main

ufoptg committed:
Merge branch 'TeamUltroid-main' into patch-1

ufoptg committed:
Merge branch 'main' into patch-1

New-dev0 commented:
Thanks for the efforts, but we currently cant consider these changes. You can re open a PR if any important fix or usefull plugin got missed
